src/runtime-rs/crates/agent/src/sock/vsock.rs

```diff
@@ -26,7 +33,50 @@ impl Vsock {
 
 #[async_trait]
 impl Sock for Vsock {
-    async fn connect(&self, _config: &ConnectConfig) -> Result<Stream> {
-        todo!()
+    async fn connect(&self, config: &ConnectConfig) -> Result<Stream> {
```

Comment:
It seems that if the connection failed once, the created socket fd is not cleaned up.

Reply:
When connection failed, no one will take unix stream's ownership. So the fd will drop with it. 
https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#167-179
<img width="794" alt="image" src="https://user-images.githubusercontent.com/118146271/202645923-9b331839-95d4-460a-b9b7-88ff1e54e175.png">

It seems fine?

Reply:
I agree 😄 . I checked the `nix` library and see
```Rust
#[stable(feature = "unix_socket", since = "1.10.0")]
impl FromRawFd for UnixStream {
    #[inline]
    unsafe fn from_raw_fd(fd: RawFd) -> UnixStream {
        UnixStream(Socket::from_inner(FromInner::from_inner(OwnedFd::from_raw_fd(fd))))
    }
}
```

Reply:
cool